clients/drcachesim/tracer/raw2trace.h

```diff
@@ -672,6 +679,103 @@ struct trace_header_t {
             return msg;    \
     } while (0)
 
+template <typename T> class bitset_hash_table_t {
+private:
+    const static size_t BLOCK_SIZE_BIT = 13;
+    const static size_t BLOCK_SIZE = (1 << BLOCK_SIZE_BIT);
+    const size_t BASIC_BUCKET_COUNT = (1 << 15);
+    std::unordered_map<T, std::bitset<BLOCK_SIZE>> page_table;
+    typename std::unordered_map<T, std::bitset<BLOCK_SIZE>>::iterator last_block =
+        page_table.end();
+
+    inline std::pair<T, ushort>
+    convert(T pc)
+    {
+        return std::pair<T, ushort>(
+            reinterpret_cast<T>(reinterpret_cast<size_t>(pc) & (~(BLOCK_SIZE - 1))),
```

Comment:
Please add comments to make it more clear what this function is doing and why it's doing it.  A more descriptive name for the function which indicates what it is converting to would help.  Creating a custom struct with named fields would also help instead of using pair which leaves us with "first" and "second" which do not help readability.

Reply:
#6176